go/consensus/cometbft/apps/vault/transactions.go

```diff
@@ -0,0 +1,218 @@
+package vault
+
+import (
+	"fmt"
+
+	"github.com/oasisprotocol/oasis-core/go/common/errors"
+	"github.com/oasisprotocol/oasis-core/go/consensus/cometbft/api"
+	stakingState "github.com/oasisprotocol/oasis-core/go/consensus/cometbft/apps/staking/state"
+	vaultState "github.com/oasisprotocol/oasis-core/go/consensus/cometbft/apps/vault/state"
+	vault "github.com/oasisprotocol/oasis-core/go/vault/api"
+)
+
+func (app *vaultApplication) create(ctx *api.Context, args *vault.Create) error {
```

Comment:
I suggested that because `authorizeAction` is a long method and at the end I had no idea where `args.Nonce` comes from. I checked `staking` app, and there you have names like `burn *staking.Burn`. Name `args` is just so general, that I would try to avoid it. Maybe `req` should do it, not sure, you decide.